.circleci/config.yml

```diff
@@ -429,21 +428,51 @@ commands:
                 name: "Halt running Playwright test job. Not supported for ENV=<< parameters.env >>"
                 command: circleci-agent step halt
 
+  halt-when-study-unchanged:
+    description: Stop job if specified study has not changed
+    parameters:
+      study_key:
+        type: string
+    steps:
+      - run:
+          name: Check << parameters.study_key >> and stop job if study has not changed
+          command: |
+            set -u
+            CHANGED_STUDIES=$(../build-utils/findmodifiedprojects.sh << pipeline.git.base_revision >> << pipeline.git.revision >>)
+            echo "The following projects were found to have changed: $CHANGED_STUDIES"
+            
+            NG_PROJECT_NAME="ddp-<< parameters.study_key >>"
+            CHANGED_DEPENDENCIES=$(echo $CHANGED_STUDIES | grep -E -e $NG_PROJECT_NAME -e _SHARED_ -e ddp-sdk -e toolkit)
+            echo "CHANGED_DEPENDENCIES: $CHANGED_DEPENDENCIES"
+            
+            if [[ -z $CHANGED_DEPENDENCIES ]]; then
+              echo "Study << parameters.study_key >> has not changed"
+              circleci-agent step halt
+            else
+              echo "Study has changed"
+            fi
+        #
 
 jobs:
   app-deploy:
     working_directory: *ng_workspace_path
     parameters:
       study_key:
         type: string
+      check_changes:
+        type: boolean
+        default: false
     executor:
       name: build-executor
     steps:
+      - checkout-code
+      - when:
+          condition: << parameters.check_changes >>
+          steps:
+            - halt-when-study-unchanged:
+                study_key: << parameters.study_key >>
```

Comment:
Inserting `checkout-code` call here (in job) because `halt-when-study-unchanged` requires ddp-angular project code.